### ISSUE TITLE
Parse grammar.xml to a set of Ruby objects

### ISSUE THREAD

**Issue by zverok** [molybdenum-99/proofreader#3]
Each language's main grammar rule set is situated at
```
languagetool-language-modules/<lang>/src/main/resources/org/languagetool/rules/<lang>/grammar.xml
```
Here is the beginning of `grammar.xml` for English (without some prelude):
```xml
<rules lang="en" xsi:noNamespaceSchemaLocation="../../../../../../../../../languagetool-core/src/main/resources/org/languagetool/rules/rules.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    
    <!-- ====================================================================== -->
    <!-- Possible typo -->
    <!-- ====================================================================== -->
    <category id="TYPOS" name="Possible Typo" type="misspelling">
        <rule id="AS_FOLLOW_AS_FOLLOWS" name="'as follow' (as follows)">
         <pattern>
          <marker>
          <token>as</token>
          <token>follow</token>
          </marker>
          <token regexp='yes'>[\.:,—\-–]</token>
         </pattern>
         <message>Did you mean <suggestion>as follows</suggestion>?</message>
         <short>Possible typo</short>
         <example correction='as follows'>We can elaborate this distinction <marker>as follow</marker>.</example>         
        </rule>
        <rule id="BY_PASSIVE_PARTICIPLE_BE" name="'by' + passive participle (be)">
```
This should be parsed into set of `Rule` objects, grouped by `Category` and `RuleGroup` objects, each rule having `id`, `name`, `pattern` (represented by `Pattern` object) and so on.

Set of rules should be easily navigable and exportable to YAML (for reading and passing to other software).

**Comment by rajastakilos** [comment]
Hi Victor. I'm taking this part of the project on first. I'm reviewing the existing work in Proofreader as well.

**Comment by zverok** [comment]
Closed with #13
